runtime/rasdump/dmpmap.c

```diff
@@ -479,6 +500,31 @@ mapDumpSwitches(J9JavaVM *vm, J9RASdumpOption agentOpts[], IDATA *agentNum)
 
 	return OMR_ERROR_NONE;
 }
+
+/*
+ * Make "-XX:-HeapDumpOnOutOfMemoryError" disable dumps on OOM
+ */
+void
+disableDumpOnOutOfMemoryError(J9JavaVM *vm, J9RASdumpOption agentOpts[], IDATA agentNum)
+{
+	int kindCursor = 0;
+
+	for (kindCursor = 0; kindCursor < numOomSettings; ++kindCursor) {
+		char *typeString = oomDefaultTable[kindCursor].typeString;
+		IDATA disableKind = scanDumpType(&typeString);
+		int j = 0;
+		for (j = 0; j < agentNum; j++) {
+			if (NULL != agentOpts[j].args) {
+				if (agentOpts[j].kind == disableKind) {
+					if (strcmp(agentOpts[j].args, oomDefaultTable[kindCursor].onOpt) == 0) {
```

Comment:
Joining these three conditions would reduce the indentation significantly.

Reply:
Good idea.  Thanks.